Net31Solution/SensorApp/Sensor.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.ComponentModel;
+
+namespace SensorApp
+{
+    internal class Sensor : INotifyPropertyChanged
+    {
+        private int _measuredValue;
+        private Mode _sensorMode;
+
+        public int MeasuredValue
+        {
+            get
+            {
+                return _measuredValue;
```

Comment:
Each time _mesureValue is get, it should be regenerated depending on current sensor mode.